chemprop/cli/utils/parsing.py

```diff
@@ -213,17 +216,39 @@ def build_data_from_files(
     smiss, rxnss, Y, weights, lt_mask, gt_mask = parse_csv(
         p_data, smiles_cols, rxn_cols, target_cols, ignore_cols, weight_col, bounded, no_header_row
     )
-    X_f = np.load(p_features) if p_features else None
-    V_fs = np.load(p_atom_feats, allow_pickle=True) if p_atom_feats else None
-    E_fs = np.load(p_bond_feats, allow_pickle=True) if p_bond_feats else None
-    V_ds = np.load(p_atom_descs, allow_pickle=True) if p_atom_descs else None
+    n_molecules = len(list(zip(*smiss)))
+
+    X_ds = load_input_features(p_features, n_molecules, feature="X_d")
+    V_fss = load_input_features(p_atom_feats, n_molecules, feature="V_f")
+    E_fss = load_input_features(p_bond_feats, n_molecules, feature="E_f")
+    V_dss = load_input_features(p_atom_descs, n_molecules, feature="V_d")
 
     mol_data, rxn_data = make_datapoints(
-        smiss, rxnss, Y, weights, lt_mask, gt_mask, X_f, V_fs, E_fs, V_ds, **featurization_kwargs
+        smiss, rxnss, Y, weights, lt_mask, gt_mask, X_ds, V_fss, E_fss, V_dss, **featurization_kwargs
     )
 
     return mol_data + rxn_data
 
+def load_input_features(paths, n_molecules, feature):
+    if paths is None:
+        return None
+
+    match feature:
+        case "X_d":
+
+            path = paths
+            loaded_feature = np.load(path)
+            features = loaded_feature["arr_0"]
+
+        case _:
+
+            features = []
+            for mol_idx in range(n_molecules):
```

Comment:
```suggestion
            for _ in range(n_molecules):
```

`mol_idx` isn't used, so here is a stylistic change for you to consider.